### PR TITLE
V3 & V5 constructors

### LINKED ISSUE #8: Constructor for creating v3 (namespaced, MD5 hash) UUIDs
Note that we'll need to provide a typeclass for converting values into a ByteArray that we can digest with MD5, see https://github.com/danlentz/clj-uuid/blob/master/src/clj_uuid.clj#L617

### PR DESCRIPTION
**Pull request by purrgrammer**
Closes #8 and #9

### CODE REVIEW

File: src/test/scala/memeid/V3Spec.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import cats.effect._
+import cats.syntax.eq._
+
+import org.specs2.ScalaCheck
+import org.specs2.matcher.IOMatchers
+import org.specs2.mutable.Specification
+
+class V3Spec extends Specification with ScalaCheck with IOMatchers {
+  "V3 constructor" should {
+    "create same UUID for the same namespace/name" in {
```
Comment (alejandrohdezma) [inline src/test/scala/memeid/V3Spec.scala]: Maybe add a test that checks the opposite (distinct UUIDs for distinct namespace/name)

File: src/test/scala/memeid/V5Spec.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import cats.effect._
+import cats.syntax.eq._
+
+import org.specs2.ScalaCheck
+import org.specs2.matcher.IOMatchers
+import org.specs2.mutable.Specification
+
+class V5Spec extends Specification with ScalaCheck with IOMatchers {
+  "V5 constructor" should {
+    "create same UUID for the same namespace/name" in {
```
Comment (alejandrohdezma) [inline src/test/scala/memeid/V5Spec.scala]: Maybe add a test that checks the opposite (distinct UUIDs for distinct namespace/name)

File: src/main/scala/memeid/Constructors.scala [hunk 3]
```diff
@@ -37,4 +44,31 @@ trait Constructors {
       })
     }
 
+  def v3[F[_]: Sync, A](namespace: UUID, local: A)(implicit D: Digestible[A]): F[UUID] =
+    Sync[F].delay {
+
+      val digest = MD5.digest
```
Comment (alejandrohdezma) [inline src/main/scala/memeid/Constructors.scala]: Both `v3` and `v5` constructors could be refactored with a method like:

```scala
def hashedUUID[A: Digestible](algorithm: Algorithm, version: Short, namespace: UUID, name: A)
```

File: src/test/scala/memeid/V5Spec.scala (hunk 2 above)
Comment (purrgrammer) [reply src/test/scala/memeid/V5Spec.scala]: Added, your comment helped catch a bug by the way :clap:

File: src/main/scala/memeid/Constructors.scala (hunk 3 above)
Comment (purrgrammer) [reply src/main/scala/memeid/Constructors.scala]: Done!

**Pull request merged**